code/range/ViewModel/src/main/java/ru/unn/agile/range/viewmodel/ViewModel.java

```diff
@@ -0,0 +1,186 @@
+
+        if (isInteger(input)) {
+            if (range.containsValue(Integer.parseInt(input))) {
+                txtResult.setValue("Yes");
+            } else {
+                txtResult.setValue("No");
+            }
+        }
+        if (isIntegerSet(input)) {
+            if (range.containsSet(Arrays.stream(input.substring(1, input.length() - 1).split(","))
+                    .map(String::trim).mapToInt(Integer::parseInt).toArray())) {
+                txtResult.setValue("Yes");
+            } else {
+                txtResult.setValue("No");
+            }
+        }
+        if (isRange(input)) {
+            if (range.containsRange(new Range(input))) {
+                txtResult.setValue("Yes");
+            } else {
+                txtResult.setValue("No");
+            }
+        }
+    }
+
+    public void overlapsRange() {
+        String input = txtInput.get();
+        if (isRange(input)) {
+            if (range.overlapsRange(new Range(input))) {
+                txtResult.setValue("Yes");
+            } else {
+                txtResult.setValue("No");
+            }
+        }
+    }
+
+    public void equalsRange() {
+        String input = txtInput.get();
+        if (isRange(input)) {
+            if (range.equals(new Range(input))) {
+                txtResult.setValue("Yes");
+            } else {
+                txtResult.setValue("No");
+            }
+        }
+    }
+
+    public void getAllPoints() {
+        txtResult.setValue(Arrays.toString(range.getAllPoints()));
+    }
+
+    public void getEndPoints() {
+        txtResult.setValue(Arrays.toString(range.endPoints()));
+    }
+
+
+    private void setRange(final String input) {
+
+        var isCorrectInput = true;
+
+        try {
+            this.range = new Range(input);
+        } catch (IllegalArgumentException ex) {
+            isCorrectInput = false;
+        }
+
+        btnGetAllPointsDisabled.setValue(!isCorrectInput);
+        btnGetEndPointsDisabled.setValue(!isCorrectInput);
+        disableAllButtonsConnectedWithInput();
+    }
+
+    private void setInput(final String input) {
+        disableAllButtonsConnectedWithInput();
+
+        if (btnGetEndPointsDisabled.get()) {
+            return;
+        }
+
+        if (isInteger(input) || isIntegerSet(input)) {
+            btnContainsDisabled.setValue(false);
+        } else if (isRange(input)) {
+            btnContainsDisabled.setValue(false);
+            btnOverlapsDisabled.setValue(false);
+            btnEqualsDisabled.setValue(false);
+        }
+    }
+
+    private void disableAllButtonsConnectedWithInput() {
+        btnContainsDisabled.setValue(true);
+        btnOverlapsDisabled.setValue(true);
+        btnEqualsDisabled.setValue(true);
+    }
+
+    private boolean isInteger(final String input) {
+        return input.matches("-?\\d+");
+    }
+
+    private boolean isIntegerSet(final String input) {
+        return input.matches("\\{-?\\d+(,-?\\d+)+?\\}");
```

Comment:
You use trim at 87, then regexp should look like this \\s*\\{-?\\d+(\\s*,\\s*-?\\d+)+?\\}\\s*

Reply:
It was done intentionally. I don't see any reason to allow range with spaces inside:)